core/modules/qdisp/QdispPool.cc

```diff
@@ -201,17 +204,46 @@ std::string PriorityQueue::statsStr() {
 }
 
 
+QdispPool::QdispPool(int poolSize, int largestPriority, std::vector<int> const& maxRunSizes,
+                      std::vector<int> const& minRunningSizes) {
+    std::stringstream os;
+    os << "poolSize=" << poolSize
+       << " maxPriority=" << largestPriority << " maxRunSizes=" << util::prettyCharList(maxRunSizes)
+       << " minRunningSizes=" << util::prettyCharList(minRunningSizes);
+    if (poolSize < 1 || largestPriority < 0 || maxRunSizes.size() < static_cast<size_t>(largestPriority) + 1) {
+        LOGS(_log, LOG_LVL_ERROR, "QdispPool invalid paramater " << os.str());
+        throw std::invalid_argument(os.str());
+    }
+
+    LOGS(_log, LOG_LVL_INFO, "QdispPool creating " << os.str());
+    _prQueue = std::make_shared<PriorityQueue>(100, 1, 1); // default (lowest) priority.
+    for (unsigned int pri = 0; pri <= static_cast<unsigned int>(largestPriority); ++pri) {
+        size_t minRun = 0;
+        if (minRunningSizes.size() > pri) {
+            minRun = minRunningSizes[pri];
+        }
+        size_t maxRun = 1;
+        if (maxRunSizes.size() > pri) {
+            maxRun = maxRunSizes[pri];
+        }
+        LOGS(_log, LOG_LVL_INFO, "creating priQ pri=" << pri << " min=" << minRun << " max=" << maxRun);
+        _prQueue->addPriQueue(pri, minRun, maxRun);
+    }
+    _pool = util::ThreadPool::newThreadPool(poolSize, _prQueue);
+}
+
+
 void QdispPool::_setup(bool unitTest) {
     if (not unitTest) {
         // Numbers are based on 1200 threads in the _pool. Large results
         // tend to be slow to give up their threads, thus can't be allowed
         // to eat up the pool. Bandwidth also makes running many of the
         // slow queries at the same time a burden on the system.
         // TODO: Set up thread pool size and queues in configuration. DM-10237
-        _prQueue = std::make_shared<PriorityQueue>(100, 1, 5); // default (lowest) priority.
+        _prQueue = std::make_shared<PriorityQueue>(100, 1, 1); // default (lowest) priority.
         _pool = util::ThreadPool::newThreadPool(2400, _prQueue);
         _prQueue->addPriQueue(0, 1, 90);  // Highest priority - interactive queries
-        _prQueue->addPriQueue(1, 1, 1000);  // Outgoing shared scan queries.
+        _prQueue->addPriQueue(1, 1, 1000); // Outgoing shared scan queries.
```

Comment:
This is the second time I'm complaining about using the hard-wired numbers in initializing parameters of `QdispPool`. I did so when I was reviewing another ticket a few months ago. It would be =the right thing to move these values into `czar`'s configuration file.

Reply:
This is hanging code and is now removed.